src/main/java/com/shapesecurity/functional/data/ConcatList.java

```diff
@@ -83,6 +85,18 @@ protected ConcatList(int length) {
   @NotNull
   public abstract Maybe<ConcatList<T>> update(int index, @NotNull T element);
 
+  @NotNull
+  public final T[] toArray(T[] array) {
+    @SuppressWarnings("unchecked")
+    T[] target = (array.length != this.length) ?
```

Comment:
That's not what it is usually designed. The array used as a carrier of the runtime type information of `T`.